mitodl/micromasters
# Skip if the status is not terminal

#### What are the relevant tickets?
Fixes #2532 

#### What's this PR do?
Instead of just checking if the user is in some financial aid state, it explicitly checks if the user is approved or skipped already if the user has a 100% off coupon for a program.

#### How should this be manually tested?
Set up financial aid for a program so that it's pending documents or otherwise waiting for input from the reviewer. Create a 100% off program coupon. Apply that coupon and verify that the financial aid status for that user for that program is now skipped.
